src/Tinker.jl

```diff
@@ -12,24 +12,40 @@ struct Rectangle <: Shape
     y::Number
     w::Number
     h::Number
+    pts::AbstractArray
 end
 
-Rectangle() = Rectangle(0,0,-1,-1)
+Rectangle() = Rectangle(0,0,-1,-1, [])
 Base.isempty(R::Rectangle) = R.w <= 0 || R.h <= 0
 
 mutable struct ImageContext{T}
     image
     canvas::GtkReactive.Canvas
     zr::Signal{ZoomRegion{T}}
     zl::Int # for tracking zoom level
-    pandrag::Signal{Bool}
-    zoomclick::Signal{Bool}
-    rectselect::Signal{Bool}
-    shape::Signal{<:Shape} # Holds selection outline
-    rectview::Signal{<:AbstractArray} # holds rectangular region corresponding to outline (type?)
+    mouseactions::Dict{String,Signal{Bool}}
+    shape::Signal{<:Shape} # Tracks type of selection in the environment
+    points::Signal{<:AbstractArray} # Holds points that define shape outline
+    rectview::Signal{<:AbstractArray} # Holds rectangular region corresponding to outline
 end
 
-ImageContext() = ImageContext(nothing, canvas(), Signal(ZoomRegion((1:10, 1:10))), -1, Signal(false), Signal(false), Signal(false), Signal(Rectangle()), Signal([]))
+ImageContext() = ImageContext(nothing, canvas(), Signal(ZoomRegion((1:10, 1:10))), -1, Dict("dummy"=>Signal(false)), Signal(Rectangle()), Signal([]), Signal([]))
+
+function get_view(image,x_min,y_min,x_max,y_max)
+    xleft,yleft = Int(floor(Float64(x_min))),Int(floor(Float64(y_min)))
```

Comment:
Do you want to use `floor` or `round`? Or perhaps `floor` for the lower and `ceil` for the upper? (or vice versa) If you do really mean `floor`, ignore this.